File: efficiency_tests/scripts/subscriber.py
#!/usr/bin/env python
import rospy
from efficiency_tests.msg import SuperMessage

publish_rate_old = 0
message_counter = 0


def callback(data):
    global publish_rate_old, message_counter, start_time
    publish_rate = rospy.get_param("/rate_value", 10)
    if(publish_rate != publish_rate_old):  # If changed
            publish_rate_old = publish_rate
            message_counter = 0

    if(len(data.data) != 0):
        message_counter = message_counter+1
        current_time = rospy.get_rostime()
        print("Message Count : " + str(message_counter))


def listener():
    rospy.init_node('subscriber_python_node')
    rospy.Subscriber("topic", SuperMessage, callback)
    rospy.spin()

if __name__ == '__main__':
    listener()

File: efficiency_tests/src/subscriber.cpp
#include "ros/ros.h"
#include "efficiency_tests/custom_msg.h"

int message_counter = 0, publish_rate_old = 0;
std::int32_t publish_rate;
void callback(const efficiency_tests::custom_msg::ConstPtr& msg)
{
  ros::param::param<std::int32_t>("/rate_value", publish_rate, 1);
  if(publish_rate != publish_rate_old)  // If changed
  {
    publish_rate_old = publish_rate;
    message_counter = 0;
  }

  if(msg->data.length() > 0)
  {
    //Means, data is not empty
    message_counter++;
    ROS_INFO("Message Count : %d" , message_counter);
  }
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "subscriber_cpp_node");
  ros::NodeHandle n;
  ros::Subscriber sub = n.subscribe("topic", 1000, callback);
  ros::spin();

  return 0;
}